stl/inc/mdspan

```diff
@@ -8,9 +8,9 @@
 #define _MDSPAN_
 #include <yvals.h>
 #if _STL_COMPILER_PREPROCESSOR
-#if !_HAS_CXX23
+#if !_HAS_CXX23 || !defined(__cpp_lib_concepts) // TRANSITION, GH-395
 _EMIT_STL_WARNING(STL4038, "The contents of <mdspan> are available only with C++23 or later.");

```

Comment:
Shouldn't this also be updated to say "with concepts support" like `<concepts>` and similarly `<source_location>`

Reply:
Good catch, but perhaps surprisingly, the variation about mentioning concepts doesn't really matter. This is because we have enabled concepts when we specifically detect IntelliSense in the IDE (as opposed to our command-line only use of `/BE`) - this is by special agreement with the IDE team that they won't complain about any bugs encountered, because IntelliSense doesn't need to fully instantiate templates in order to provide most of its functionality. So in practice, users don't experience "C++23 without concepts" mode, only we do in the test suite.

Reply:
Exact precedent: https://github.com/microsoft/STL/blob/daa994bfc41c36196c536f2b68388f859d6bd656/stl/inc/expected#L11-L13